src/main/java/thePackmaster/powers/bladestormpack/WindrushPower.java

```diff
@@ -0,0 +1,52 @@
+package thePackmaster.powers.bladestormpack;
+
+import com.megacrit.cardcrawl.actions.common.ReducePowerAction;
+import com.megacrit.cardcrawl.actions.common.RemoveSpecificPowerAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import thePackmaster.powers.AbstractPackmasterPower;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+
+//REFS: BufferPower & TungstenRod relic (base game), ConstructPower (entropypack)
+public class WindrushPower extends AbstractPackmasterPower {
+    public static final String POWER_ID = makeID("WindrushPower");
+    private static final String NAME = CardCrawlGame.languagePack.getPowerStrings(POWER_ID).NAME;
+    private static final String[] DESCRIPTIONS = CardCrawlGame.languagePack.getPowerStrings(POWER_ID).DESCRIPTIONS;
+
+    public WindrushPower(AbstractCreature owner, int amount) {
+        super(POWER_ID, NAME, PowerType.BUFF, true, owner, amount);
+
+        this.priority = -42068; //ConstructPower: -42069, BufferPower: 5. Triggers after Construct, and protects Buffer.
+    }
+
+    //Based on BufferPower, which does not affect enemy intents (that results in over-mitigation).
+    @Override
+    public int onAttackedToChangeDamage(DamageInfo info, int damageAmount) {
+        if (info.type != DamageInfo.DamageType.NORMAL) { return damageAmount; }
+
+        //Block DID already reduce damage by this point. Game pretends blocked damage never happened.
+        if (damageAmount > 0) {    //So this is only unblocked damage.
+            this.flashWithoutSound();
+            return Math.max(0, damageAmount - this.amount);
+        }
+        return damageAmount;
+    }
+
+    @Override
+    public void atEndOfRound() {
+        //Cut in half, rounding down ( = rounding the reduction up) and able to fully fade out.
+        if (amount == 1) {
+            addToBot(new RemoveSpecificPowerAction(this.owner, this.owner, this));
+        } else if ((amount % 2) != 0){
+            addToBot(new ReducePowerAction(this.owner, this.owner, this, (amount+1) / 2));
```

Comment:
You can simplify this function down to only this single line. With integers, `(amount + 1) / 2` produces the correct result for both even and odd numbers, and `ReducePowerAction` handles removing the power if the reduction is greater than or equal to the current amount of the power. So this one case handles it all and the other cases are unnecessary.